views/hooks/useChartsList.jsx

```diff
@@ -69,16 +69,18 @@ export default function useChartsList() {
       setNotification({ open: true, message: error.message })
     }
   }
-  const loadCharts = async () => {
+  const loadCharts = async (queryParams = {}) => {
     try {
-      const data = await api.charts.chart.index()
+      const data = await api.charts.chart.all(queryParams)
```

Comment:
do we need the `charts.chart` part? Seems duplicative. Can we make this `api.charts.all(queryParams)`?

Reply:
The api.charts is an object of the different endpoints we want to trigger for this controller.  The api object and the corresponding routes are listed.

```javascript
const charts = {
  chart,
  chartsData,
  chartsDuplicate,
  chartsShare,
}

//routes file
  chart: {
    show: (chart_id) => `${apiPath}/charts/${chart_id}`,
    all: `${apiPath}/charts`,
  },
  shareChart: {
    show: (chart_id) => `${apiPath}/charts/${chart_id}/share`,
  },
  duplicateChart: {
    show: `${apiPath}/charts/duplicate`,
  },
```
The endpoints on the server are slightly different, but I can change it to api.charts.chart.all.

Reply:
👍 I was referring to the methods we're calling here, so that's fine leaving it as-is. It's out of scope of this PR but I think we can clean this up so we're not nesting these objects. For example, I think

```js
api.charts.show(chartID)
api.charts.share(chartID)
api.charts.duplicate(chartID)
api.charts.all()
```

is cleaner than

```js
api.charts.chart.show(chartID)
api.charts.chartShare.create(chartID)
api.charts.chartsDuplicate.create(chartID)
api.charts.chart.all()
```

Reply:
yes and that does look cleaner, we have opportunities to refactor and will continue to make updates like these as we progress 👍